### PR TITLE
Chrono Grenades

### PR DESCRIPTION
**Pull request by Shadowmech88**
Adds the chrono grenade and box of chrono grenades. Both currently unobtainable.
Chrono grenades, upon detonation, generate a time stop field in a five-tile radius around them for ten seconds.

Adds a general proc that can be used to generate a time stop field on a given atom, with a given duration and range.

Adds `timeless` flag, which can be applied to atoms in order to render the atom immune to time manipulation.
Adds a `toggle_timeless()` proc that can be called to toggle an atom's `TIMELESS` flag on or off.

This is the first of a small series of time manipulation items I plan to add.

:cl:
 * rscadd: Added chrono grenades, which stop time in a five-tile radius for ten seconds upon detonation.

### CODE REVIEW

**Comment by clusterfack** [comment]
Neat idea except for the new atom var for a boolean

**Comment by Shadowmech88** [comment]
Should I make it an atom-level proc instead?

**Comment by Shadowmech88** [comment]
Actually I'd rather not, since that'd mean things couldn't be varedited to be time-immune.

What other alternatives are there?

**Comment by Exxion** [comment]
Flag

**Comment by Shadowmech88** [comment]
Is it feasible for admins to add flags to an atom on the fly? I'm not super well-versed in VV.

**Comment by Exxion** [comment]
Yes, though it's slightly more difficult than just flipping a var

**Comment by Shadowmech88** [comment]
Where should I put the flag define? I heard that we shouldn't add anything else to `setup.dm`.
Should I make a new file in `__DEFINES` just for this flag?

**Comment by PJB3005** [comment]
If you're putting them in a flags var that has other flags in `setup.dm` using `setup.dm` is fine to add to that list, otherwise make a new file yes.

**Comment by Shadowmech88** [comment]
I made it a flag *and* added a proc that admins can manually call on an atom in order to give/take the flag.

File: code/game/atoms.dm [hunk 1]
```diff
@@ -805,3 +804,11 @@ its easier to just keep the beam vertical.
 //Called when loaded by the map loader
 /atom/proc/spawned_by_map_element(datum/map_element/ME, list/objects)
 	return
+
+/atom/proc/toggle_timeless()
```
Comment (Exxion) [inline code/game/atoms.dm:808]: `flags ^= TIMELESS`
I'm pretty sure ^= is a valid operator at least

Comment (clusterfack) [inline code/game/atoms.dm]: It is but it isn't what you want to toggle a specific flag

Comment (Exxion) [reply code/game/atoms.dm:808]: Why not?

**Comment by Shadowmech88** [comment]
> [Needs Changelog]

Why? None of this is obtainable yet outside of adminbus.
Or do you mean so that admins know they exist?

**Comment by Probe1** [comment]
for admins ya

**Comment by Shadowmech88** [comment]
Changelog done.

File: code/game/atoms.dm (hunk 1 above)
Comment (PJB3005) [reply code/game/atoms.dm:808]: Seems like it'd work to me.

Comment (Exxion) [reply code/game/atoms.dm:808]: @Shadowmech88 Cluster was confused
`^=` works for toggling

**Pull request merged**